src/plugins/main.js

```diff
@@ -148,13 +169,7 @@ export default async (context) => {
   app.i18n.onLanguageSwitched = <%= options.onLanguageSwitched %>
   app.i18n.setLocaleCookie = setLocaleCookie
   app.i18n.getLocaleCookie = getLocaleCookie
-  app.i18n.setLocale = async locale => {
-    await loadAndSetLocale(locale)
-
-    if (useCookie) {
-      setLocaleCookie(locale)
-    }
-  }
+  app.i18n.setLocale = loadAndSetLocale
```

Comment:
Should public API have access to `init` parameter? I suppose not in which case it would be better to do:
```js
app.i18n.setLocale = locale => loadAndSetLocale(locale)

Reply:
Actually arrow function might not work so:
```js
app.i18n.setLocale = function(locale) { return loadAndSetLocale(locale); }
```